conda/plugins/manager.py

```diff
@@ -226,7 +226,7 @@ def disable_external_plugins(self) -> None:
         Disables all currently registered plugins except built-in conda plugins
         """
         for name, plugin in self.list_name_plugin():
-            if not name.startswith("conda.plugins."):
+            if not name.startswith("conda.plugins.") and not self.is_blocked(name):
```

Comment:
Just in case we're running `disable_external_plugins` a few times, we can make sure that we don't run expensive code to block something if it's already being blocked.